### PR TITLE
Feature/56 weather server component

### LINKED ISSUE #56: Basic weather comopnent

### PR DESCRIPTION
**Pull request by Derstilon**
Closes #56

### CODE REVIEW

**Comment by retkiewi** [comment]
Add this node to the "start" scripts.

**Review by retkiewi — CHANGES_REQUESTED** [review]

File: weather_server/main.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+	weather_type = json_data.get("options").get("weather_type")
+	if weather_type not in LABELS:
+		return make_response("Invalid weather type specified", 400)
+
+class WeatherModule(Resource):
+	"""Class responsible for handling the weather module"""
+ 
+	@staticmethod
+	def get_weather(weather_type, precision, paths, results, index, size=1):
+		"""Get weather"""
```
[the remark this answers is not in the document]
Comment (Derstilon) [reply weather_server/main.py]: done. Can be checked here:
https://github.com/ostatni5/tk-2022/blob/8aa56c03cc7c9d79a77e5b7f73ff1869dd6979b5/weather_server/weather_module.py#L31

File: weather_server/weather_module.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+			return make_response(response_err, 400)
+
+		paths = json_data.get("paths")
+		threads = []
+		filter_array = [False] * len(paths)
+		weather_type = json_data.get("options").get("weatherType")
+		precision = json_data.get("options").get("precision", 0)
+		
+		#split paths into chunks
+		chunk_size = len(paths) // THREAD_LIMIT + (1 if len(paths) % THREAD_LIMIT != 0 else 0)
```
Comment (retkiewi) [inline weather_server/weather_module.py:66]: The chunking and filtering should probably be separate functions.

File: weather_server/__main__.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+from create_app import create_app
+
+def main():
+	create_app().run(host='localhost', port=8086)
```
Comment (retkiewi) [inline weather_server/__main__.py:4]: Wrong indent.

File: weather_server/check_json.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
+from flask import make_response
+
+LABELS = []
+
+weather_types_file = open("./predict_weather_in_image/retrained_labels.txt", "r")
+
+for line in weather_types_file:
```
Comment (retkiewi) [inline weather_server/check_json.py:7]: Use list comprehensions -> LABELS = [<list comprehension>]

File: weather_server/create_app.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
+from flask import Flask
+from flask_cors import CORS
+from flask_restful import Api
+from weather_module import WeatherModule
+
+def create_app():
+	app = Flask(__name__)
```
Comment (retkiewi) [inline weather_server/create_app.py:7]: wrong indent

File: weather_server/test/test_module.py [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+from create_app import create_app
+from weather_module import WeatherModule
+from urllib.request import urlopen
+from flask import request
+import os
+ROOT_PATH = os.getcwd().split("weather_server")[0].replace(os.sep, '/')
+
+class TestModule:
+    paths = [
+        os.path.join(ROOT_PATH,"resources/Cloudy/Cloudy13.jpg"),
```
Comment (retkiewi) [inline weather_server/test/test_module.py:14]: Indent formatting

File: weather_server/weather_module.py [hunk 7]
```diff
@@ -0,0 +1,94 @@
+from marshmallow import Schema, fields
```
Comment (retkiewi) [inline weather_server/weather_module.py:1]: This file has different indents than the rest.

**Review by retkiewi — CHANGES_REQUESTED** [review]
Almost there :)

File: weather_server/check_json.py [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+		error = "No options specified"
+
+	if not json_data.get("options").get("weatherType"):
+		error = "No weather type specified"
+		
+	weather_type = json_data.get("options").get("weatherType")
+	if weather_type not in LABELS:
+		error = f"Invalid weather type: {weather_type}"
+
+	return error
```
Comment (retkiewi) [inline weather_server/check_json.py:24]: Guess whats missing (spoiler it should be at the end of the file)

File: weather_server/create_app.py (hunk 5 above)
Comment (retkiewi) [reply weather_server/create_app.py:7]: The indent is still wrong btw

File: weather_server/test/test_module.py [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+        "rain fog"
+    ]
+        
+    weather_type = "mostlycloudy"
+
+    @pytest.fixture()
+    def app(self):
+        app = create_app()
+        app.config.update({"TESTING": True})
+        # app.run()
```
[the remark this answers is not in the document]
Comment (retkiewi) [reply weather_server/test/test_module.py:38]: .

File: weather_server/weather_module.py (hunk 7 above)
Comment (retkiewi) [reply weather_server/weather_module.py:1]: .

File: weather_server/weather_module.py (hunk 2 above)
Comment (retkiewi) [reply weather_server/weather_module.py:66]: .

**Review by retkiewi — APPROVED** [review]
Looks good :)

**Review by retkiewi — APPROVED** [review]

**Pull request merged**